rholang/src/main/scala/coop/rchain/rholang/interpreter/Reduce.scala

```diff
@@ -82,6 +78,15 @@ class DebruijnInterpreter[M[_], F[_]](
 
   type Application = Option[(TaggedContinuation, Seq[ListParWithRandom], Int, Boolean)]
 
+  private def runAndReportError(process: M[Unit]): M[Unit] =
+    errorRef.get >>= {
```

Comment:
Since `Ref` doesn't give means of synchronization I can imagine cases where identical paths of Rholang execution raise different errors. Using `MVar` for synchronization on the other hand would most probably slow reduction down dramatically. I assume we can live in a world where errors are not necessarily raised deterministically?

Reply:
I'm stil not convinced that this is enough to prevent tuplespace pollution. I also disagree with the claim that performance would suffer - this is easy to benchmark, and should be.

I'll have a look at the PR as soon as I can.

Reply:
This error model will not work because it’s not replay-able. If there are two processes in parallel composition, both of which throw errors, the order in which they’re run will determine which error is reported. That’s why the reducer didn’t cancel all threads when one erred. This is true regardless of whether we synchronize around the error reference.